backend/pkg/api/handlers/handlers_test.go

```diff
@@ -0,0 +1,288 @@
+	assert.NotEqual(t, 0, len(respData.Data.PremiumProducts), "PremiumProducts should not be empty")
+}
+
+func TestInternalGetLatestStateHandler(t *testing.T) {
+	code, body := ts.get(t, "/api/i/latest-state")
+	assert.Equal(t, http.StatusOK, code)
+
+	respData := api_types.InternalGetLatestStateResponse{}
+	err := json.Unmarshal([]byte(body), &respData)
+	assert.Nil(t, err, "error unmarshalling response")
+	assert.NotEqual(t, uint64(0), respData.Data.LatestSlot, "latest slot should not be 0")
+	assert.NotEqual(t, uint64(0), respData.Data.FinalizedEpoch, "finalized epoch should not be 0")
+}
+
+func TestInternalLoginHandler(t *testing.T) {
+	// login with email in wrong format
+	code, body := ts.post(t, "/api/i/login", bytes.NewBuffer([]byte(`{"email": "admin", "password": "admin"}`)))
+	assert.Equal(t, http.StatusBadRequest, code)
+	resp := ts.parseErrorResonse(t, body)
+	assert.Equal(t, "email: given value 'admin' has incorrect format", resp.Error, "unexpected error message")
+
+	// login with correct user and wrong password
+	code, body = ts.post(t, "/api/i/login", bytes.NewBufferString(`{"email": "admin@admin.com", "password": "wrong"}`))
+	assert.Equal(t, http.StatusUnauthorized, code, "login should not be successful")
+	resp = ts.parseErrorResonse(t, body)
+	assert.Equal(t, "unauthorized: invalid email or password", resp.Error, "unexpected error message")
+
+	// login with correct user and password
+	code, _ = ts.post(t, "/api/i/login", bytes.NewBufferString(`{"email": "admin@admin.com", "password": "admin"}`))
+	assert.Equal(t, http.StatusOK, code, "login should be successful")
+
+	// check if user is logged in and has a valid session
+	code, body = ts.get(t, "/api/i/users/me")
+	assert.Equal(t, http.StatusOK, code, "call to users/me should be successful")
+
+	meResponse := &api_types.InternalGetUserInfoResponse{}
+	err := json.Unmarshal([]byte(body), meResponse)
+	assert.Nil(t, err, "error unmarshalling response")
+	// check if email is censored
+	assert.Equal(t, meResponse.Data.Email, "a***n@a***n.com", "email should be a***n@a***n.com")
+
+	// check if logout works
+	code, _ = ts.post(t, "/api/i/logout", bytes.NewBufferString(``))
+	assert.Equal(t, http.StatusOK, code, "logout should be successful")
+
+	// check if user is logged out
+	code, _ = ts.get(t, "/api/i/users/me")
+	assert.Equal(t, http.StatusUnauthorized, code, "call to users/me should be unauthorized")
+}
+
+func TestInternalSearchHandler(t *testing.T) {
+	// search for validator with index 5
+	code, body := ts.post(t, "/api/i/search", bytes.NewBufferString(`{"input":"5","networks":[17000],"types":["validators_by_deposit_ens_name","validators_by_deposit_address","validators_by_withdrawal_ens_name","validators_by_withdrawal_address","validators_by_withdrawal_credential","validator_by_index","validator_by_public_key","validators_by_graffiti"]}`))
```

Comment:
I'd prefer if we properly formatted these long JSON strings for better readability.